src/early-types.lisp

```diff
@@ -164,6 +164,14 @@ Signals an error if FOREIGN-TYPE is undefined."))
   (:documentation
    "Return the size in bytes of a foreign type."))
 
+(define-compiler-macro foreign-type-size (&whole form foreign-type)
+  ;; Don't pass &environment to CONSTANTP or MACROEXPAND as it may make
+  ;; surprising behavior with LOAD-TIME-VALUE.
+  (if-let ((comptime-value (when (constantp (macroexpand foreign-type))
+                             (ignore-errors (eval form)))))
```

Comment:
`ignore-errors` + `eval` looks like a pretty big code smell to me; wouldn't `(locally (declare (notinline foreign-type-size)) (foreign-type-size foreign-type))` work better? It should ensure that the function is called at compilation time without recursively invoking the compiler macro again.

Reply:
Oh, nice! This was what I was looking for before I gave up on `(l-t-v ,form)` -> eval.